.github/workflows/tests.yaml

```
@@ -32,57 +29,55 @@ jobs:
           sudo snap install --no-wait shellcheck
           echo "::endgroup::"
           echo "::group::pip install"
-          python -m pip install -U .[dev,lint,types]
+          python -m pip install 'tox>=4' tox-gh
+          echo "::endgroup::"
+          echo "::group::Create virtual environments for linting processes."
+          tox run -m lint --notest
           echo "::endgroup::"
           echo "::group::Wait for snap to complete"
           snap watch --last=install
           echo "::endgroup::"
-      - name: Run black
-        run: |
-          make test-black
-      - name: Run ruff
-        run: |
-          make test-ruff
-      - name: Run codespell
-        run: |
-          make test-codespell
-      - name: Run mypy
-        run: |
-          make test-mypy
-      - name: Run pyright
-        run: |
-          make test-pyright
-      - name: Run shellcheck
-        run: |
-          make test-shellcheck
-      - name: Run yamllint
-        run: |
-          make test-yaml
-
+      - name: Run Linters
+        run: tox run -m lint
   unit-tests:
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: ["3.8", "3.10", "3.11", "3.12-dev"]
-    runs-on: ${{ matrix.os }}
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-      - name: Set up Python on ${{ matrix.platform }}
+      - name: Set up Python versions on ${{ matrix.platform }}
         uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: |
+            3.8
+            3.9
+            3.10
+            3.11
+            3.12-dev
       - name: Configure environment
         run: |
           echo "::group::pip install"
-          python -m pip install -U .[dev]
-          python -m pip install -U -e .
+          python -m pip install 'tox>=4.6'
           echo "::endgroup::"
-      - name: Run unit tests
-        run: make test-units
-
+          mkdir -p results
+      - name: Setup Tox environments
+        run: tox run -m unit-tests --notest
+      - name: Test with tox
+        run: tox run-parallel --parallel all --parallel-no-spinner --skip-pkg-install --result-json results/tox-${{ matrix.platform }}.json -m unit-tests --no-list-dependencies --notest -- --no-header --quiet -rN
```

Looks like this came from an older version of starbase. I've moved away from `run-parallel` because it can cause some weird breakages with things like coverage.
```suggestion
        run: .tox/.tox/bin/tox run --skip-pkg-install --no-list-dependencies --result-json results/tox-${{ matrix.platform }}.json -m unit-tests
```

---

Oo yes, I pulled in this improvement and a few others from the latest copy of starbase.